contracts/TokenListManager.sol

```diff
@@ -14,48 +14,43 @@ contract TokenListManager is ITokenListManager, AccessControl {
     using EnumerableSet for EnumerableSet.AddressSet;
 
     // Define roles for administrative control and user access.
-    bytes32 public constant ADMIN_ROLE = keccak256("ADMIN_ROLE");
-    bytes32 public constant USER_ROLE = keccak256("USER_ROLE");
+    bytes32 public constant TOKEN_MANAGER_ROLE = keccak256("TOKEN_MANAGER_ROLE");
 
     // Mappings for whitelisted and blacklisted tokens, categorized by token type.
     mapping(TokenType => EnumerableSet.AddressSet) private whitelistedTokens;
     mapping(TokenType => EnumerableSet.AddressSet) private blacklistedTokens;
 
-    // Contract constructor setting up initial admin role.
-    constructor() {
-        _grantRole(DEFAULT_ADMIN_ROLE, msg.sender);
-        _setRoleAdmin(USER_ROLE, ADMIN_ROLE);
-    }
-
-    // Modifier to restrict function access to users with the admin role.
-    modifier onlyAdmin() {
-        require(hasRole(ADMIN_ROLE, msg.sender), "Not authorized: caller is not an admin");
-        _;
-    }
-
     // Implement the addTokens method and add tokens to whitelist, only accessible by admin
-    function addTokens(address[] calldata tokens, TokenType tokenType) external onlyAdmin {
+    function addTokens(address[] calldata tokens, TokenType tokenType) external {
+        require(hasRole(TOKEN_MANAGER_ROLE, msg.sender), "Caller is not a token manager");
```

Comment:
How about to use the existing `hasRole` modifier in `AccessControl`?

Reply:
Solved